cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaClusterRebalanceAssemblyOperator.java

```diff
@@ -514,7 +547,13 @@ public void onClose(KubernetesClientException e) {
         if (rebalanceAnnotation == RebalanceAnnotation.none) {
             log.info("{}: Arming Cruise Control rebalance user task status timer", reconciliation);
             String sessionId = clusterRebalance.getStatus().getSessionId();
+            AtomicInteger ccAPIErrorCount = new AtomicInteger();
             vertx.setPeriodic(REBALANCE_POLLING_TIMER_MS, t -> {
+                // Check that we have not already failed to contact the API beyond the allowed number of times.
+                if (ccAPIErrorCount.get() >= MAX_API_RETRIES) {
+                    vertx.cancelTimer(t);
+                    p.fail("Unable to reach Cruise Control API after: " + MAX_API_RETRIES + " attempts");
```

Comment:
should we fail with an exception to have the caller method updating the status with that exception and moving the state machine to NotReady? Even a `RuntimeException` with this message could be ok for now. I am not sure that the status is updated accordingly if we fail with a message/string and not an exception.